### PR TITLE
Changes the code to pass testing with Crystal-v0.23.1.

### PR DESCRIPTION
**Pull request by tobyapi**
I updated the code to pass testing with Crystal-v0.22.0.

### CODE REVIEW

File: src/docker/client.cr [hunk 1]
```diff
@@ -32,7 +32,7 @@ module Docker
         HTTP::Client.unix(@url.to_s.sub(/^unix:\/\//, ""))
       elsif verify_tls?
         c = HTTP::Client.new(@url.host.not_nil!, @url.port.not_nil!, true)
-        c.ssl_context = ssl_context
+        #c.ssl_context = ssl_context
```
Comment (jeromegn) [inline src/docker/client.cr]: If the `ssl_context` isn't used anywhere, might as well delete the method defining it entirely.

Why comment it out though?

Comment (tobyapi) [reply src/docker/client.cr]: Sorry, it's my mistake. I had thought the ssl_context= is removed in v0.22.0, but I found the setter is defined in core_ext/http/client.cr.

**Comment by tobyapi** [comment]
I changed the code to use ssl_context, and remove an initialize method from SSL::Context in core_ext/openssl/ssl/context.cr because SSL::Context already has the same method.

**Comment by jeromegn** [comment]
I'm sorry @TobiasGSmollett I completely forgot about this PR. I'm happy to merge it if you still think this is a good change.

**Pull request merged**